### PR TITLE
Module structure improvement

### PR DESCRIPTION
**Pull request by dmytro-grankin**
This PR improves the module structure.

Summary:
- `cli-app` module was renamed to `todo-cli`. The module contains CLI views and clients.
- `local-inmem` module was created. The module contains a local server using `InMemoryStorageFactory`.
- A server from `todo-cli` was moved to `local-jdbc`. The module contains a local server using `MySQL` data base.

### CODE REVIEW

**Comment by dmytro-grankin** [comment]
@armiol PTAL.

File: deployment/local-jdbc/src/main/java/io/spine/examples/todolist/server/LocalJdbcServer.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+ * and the username and password should be correct.
+ *
+ * <p>The server uses the
+ * {@linkplain io.spine.client.ConnectionConstants#DEFAULT_CLIENT_SERVICE_PORT default port}.
+ *
+ * @author Dmytro Grankin
+ */
+public class LocalJdbcServer {
+
+    private static final String DB_URL_PREFIX = "jdbc:mysql:";
```
Comment (armiol) [inline deployment/local-jdbc/src/main/java/io/spine/examples/todolist/server/LocalJdbcServer.java]: Please extract the configuration values to a `.properties` file and put that to resources.

File: deployment/local-jdbc/src/main/java/io/spine/examples/todolist/server/LocalJdbcServer.java [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+        log().info("Username: {}", username);
+
+        config.setPassword(password);
+        log().info("Password: {}", password);
+
+        final DataSource dataSource = new HikariDataSource(config);
+        return dataSource;
+    }
+
+    private static String getDbUrl(String dbName) {
```
Comment (armiol) [inline deployment/local-jdbc/src/main/java/io/spine/examples/todolist/server/LocalJdbcServer.java]: Some comments are required.

File: settings.gradle [hunk 3]
```diff
@@ -2,7 +2,14 @@ include 'api-java'
...
 include 'cli-core'
 include 'testutil-api'
 include 'testutil-cli'
+
+include ':local-inmem'
+project(':local-inmem').projectDir = new File('./deployment/local-inmem')
+
+include ':local-jdbc'
+project(':local-jdbc').projectDir = new File('./deployment/local-jdbc')
+
```
Comment (armiol) [inline settings.gradle]: Please remove this line if possible.

File: deployment/local-jdbc/src/main/java/io/spine/examples/todolist/server/LocalJdbcServer.java [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+
+    private LocalJdbcServer() {
+        // Prevent instantiation of this class.
+    }
+
+    public static void main(String[] args) throws IOException {
+        final String[] actualArguments;
+        if (DEFAULT_ARGUMENTS.length != args.length) {
+            log().info("The specified arguments don't match the required format. " +
+                               "Default arguments will be used.");
```
Comment (armiol) [inline deployment/local-jdbc/src/main/java/io/spine/examples/todolist/server/LocalJdbcServer.java]: Within this message you should
* print the required format;
* print out the default arguments, if you are using those.

File: deployment/local-jdbc/src/main/java/io/spine/examples/todolist/server/LocalJdbcServer.java (hunk 1 above)
Comment (dmytro-grankin) [reply deployment/local-jdbc/src/main/java/io/spine/examples/todolist/server/LocalJdbcServer.java]: Done.

File: deployment/local-jdbc/src/main/java/io/spine/examples/todolist/server/LocalJdbcServer.java (hunk 4 above)
Comment (dmytro-grankin) [reply deployment/local-jdbc/src/main/java/io/spine/examples/todolist/server/LocalJdbcServer.java]: Done.

File: deployment/local-jdbc/src/main/java/io/spine/examples/todolist/server/LocalJdbcServer.java (hunk 2 above)
Comment (dmytro-grankin) [reply deployment/local-jdbc/src/main/java/io/spine/examples/todolist/server/LocalJdbcServer.java]: Done.

File: settings.gradle (hunk 3 above)
Comment (dmytro-grankin) [reply settings.gradle]: Done.

**Comment by dmytro-grankin** [comment]
@armiol PTAL.

File: deployment/local-inmem/src/main/java/io/spine/examples/todolist/server/LocalInMemoryServer.java [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+
+import java.io.IOException;
+
+import static io.spine.client.ConnectionConstants.DEFAULT_CLIENT_SERVICE_PORT;
+
+/**
+ * A local {@link Server} using
+ * {@link io.spine.server.storage.memory.InMemoryStorageFactory InMemoryStorageFactory}.
+ *
+ * <p>The server uses the
```
Comment (armiol) [inline deployment/local-inmem/src/main/java/io/spine/examples/todolist/server/LocalInMemoryServer.java]: I would say
```
The server exposes its {@linkplain ... gRPC API} at {@linkplain ... default port}.
```

File: deployment/local-jdbc/src/main/java/io/spine/examples/todolist/server/LocalJdbcServer.java [hunk 6]
```diff
@@ -0,0 +1,171 @@
...
+ * <p>To run the server from a command-line run the command as follows:
+ * {@code gradle :local-jdbc:runServer -Pconf=db_name,username,password}
+ *
+ * <p>If the parameters were not specified to a command or the server was ran directly,
+ * {@linkplain #getDefaultArguments() default arguments} will be used.
+ *
+ * <p>As the server uses {@code MySQL}, the data base with the specified name should be created
+ * and the username and password should be correct.
+ *
+ * <p>The server uses the
```
Comment (armiol) [inline deployment/local-jdbc/src/main/java/io/spine/examples/todolist/server/LocalJdbcServer.java]: Same as above.

File: deployment/local-jdbc/src/main/resources/data_base.properties [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+# OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL,
+# SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT
+# LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+# DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+# THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+# (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+# OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#
+
+# Default configuration properties for the data base.
```
Comment (armiol) [inline deployment/local-jdbc/src/main/resources/data_base.properties]: `database` is a single word. 

Same applies to the name of this file, which BTW I would rename to `jdbc-storage.properties`.

Also please mention that this is a local server somehow.

**Pull request merged**